src/components/TimelineRender/hook/useLastScroll.ts

```diff
@@ -0,0 +1,53 @@
+import { DbEvent } from 'core/db/schema';
+import { useState, useEffect } from 'react';
+import { useDebounce } from 'usehooks-ts';
+import { scrollHeightCache } from 'core/cache/query';
+
+export interface LastScrollProp {
+  queryCacheId: string;
+  msgList: DbEvent[];
+}
+
+export function useLastScroll({ queryCacheId, msgList }: LastScrollProp) {
+  const scrollHeight = useScrollHeight();
+  useRestoreScrollHeight(scrollHeight, queryCacheId, msgList.length > 0);
+}
+
+export const useScrollHeight = (debounceDelay = 700) => {
+  const [scrollValue, setScrollValue] = useState(0);
+  const debounceScrollValue = useDebounce(scrollValue, debounceDelay);
+
+  useEffect(() => {
```

Comment:
I recommended using [useSyncExternalStore ](https://react.dev/reference/react/useSyncExternalStore) instead of using useEffect to listen for events.

Like:
```typescript

function getSnapshot() {
  return window.scrollY;
}

function subscribe(callback) {
  window.addEventListener('scroll', callback);
  return () => {
    window.removeEventListener('scroll', callback);
  };
}

const useScrollHeight = (debounceDelay = 700) => {
  const scrollValue = useSyncExternalStore(subscribe, getSnapshot, () => 0)
  const debounceScrollValue = useDebounce(scrollValue, debounceDelay);
  return debounceScrollValue
}
```